### ISSUE TITLE
Indent Typed Racket lambda result type when on its own line

### ISSUE THREAD

**Issue by greghendershott** [greghendershott/racket-mode#516]
e.g.

```racket
(lambda ([arg : Number]) : Number
        10)
```

should be

```racket
(lambda ([arg : Number]) : Number
  10)
```

As requested by @capfredf via Racket Slack.

**Comment by greghendershott** [comment]
Using `defun` indentation style for `lambda` -- e.g.  `(put 'lambda 'racket-indent-function 'defun)` -- seems to work better. It handles the reported example:

```racket
(lambda ([arg : Number]) : Number
  10)
```

And so I think that's probably fine as a work-around @capfredf if you want to try that yourself? [Docs](https://www.racket-mode.com/#racket_002dindent_002dline).

---

Note that it's not necessarily ideal with the parameters and result type spread over multiple lines.

It's seems fine to me when you keep the result type on the same line as the last parameter:

```racket
(lambda ([a : Number]
         [b : Number]) : Number
  10)
```

But if you put the result type on its own line, then it gets indented with the function body:

```racket
(lambda ([a : Number]
         [b : Number])
  : Number
  10)
```

On the one hand, that's exactly what DrRacket does, too, when I try in 7.8. 

On the other hand, that doesn't seem ideal. Instead maybe it should be something like:

```racket
(lambda ([a : Number]
         [b : Number])
        : Number
  10)
```

?

If so, that will require writing a little custom Elisp.

---

**TL;DR**:

1. Please try `(put 'lambda 'racket-indent-function 'defun)`. You can just `M-x eval-expression` and enter that; you don't even need to add it to your init file until you confirm it's OK.

2. Assuming that works fine for you (please let me know), I'll make that change the default.

3. I'll leave this issue open, and re-title it as being about indenting the return type on its own line.

4. Someday/maybe I'll close that by doing some better thing.

**Comment by greghendershott** [comment]
Thank you for confirming! I merged a commit to make that the new default.

Now it's consistent with Dr Racket.

I re-titled this issue to be about going further -- to better handle indenting the result type when it is on its own line. Someday/maybe.

**Comment by greghendershott** [comment]
I went ahead and merged a commit to handle the result type on its own line.

I didn't spend a ton of time on it. Maybe unwise: Most times I fiddle with indentation, I feel like I'm in _Harry Potter and the Chamber of Unintended Consequences_, or its sequel, _Harry Potter and the Rug Where You Push Down One Bump and It Pops Up Elsewhere_. Fortunately there is always the famous epilog, `git revert`.
